docs/sphinx/manual/troubleshooting.rst

```diff
@@ -0,0 +1,48 @@
+..
+    Copyright (C) 2019 Mikael Simberg
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+.. _troubleshooting:
+
+===============
+Troubleshooting
+===============
+
+This section contains commonly encountered problems when compiling or using HPX.
+
+.. _troubleshooting_program_options:
+
+``Undefined reference to boost::program_options``
+=================================================
+
+Boost.ProgramOptions is not ABI compatible between all C++ versions and compilers.
+Because of this you may see linker errors similar to this:
+
+.. code-block:: text
+
+   ...: undefined reference to `boost::program_options::operator<<(std::ostream&, boost::program_options::options_description const&)'
+
+if you are not linking to a compatible version of Boost.ProgramOptions. We
+recommend that you always build |boost|_ libraries and |hpx| with the same
+compiler and C++ standard.
```

Comment:
Update this with regards to the new `hpx::program_options` module?

Reply:
Yes, my memory is indeed short...

Reply:
Done.